File: Python/1_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        previous_map = {}
        for i, n in enumerate(nums):
            if target - nums[i] in previous_map:
                return [previous_map[target - nums[i]], i]
            previous_map[n] = i
        return []

File: C++/1_TwoSum.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> previousMap;

        for (int i = 0; i < nums.size(); i++) {
            if (previousMap.find(target - nums[i]) != previousMap.end()) {
                return {previousMap[nums[i]], i};
            }
            previousMap.insert({nums[i], i});
        }
        return {};
    }
};